src/middleware/insights.ts

```diff
@@ -0,0 +1,114 @@
+import { InsightsClient, Middleware } from '../types';
+import { getInsightsAnonymousUserToken } from '../helpers';
+import { warning, noop } from '../lib/utils';
+
+export type InsightsProps = {
+  insightsClient: false | InsightsClient;
```

Comment:
Accepting `false` for the Insights client is confusing because it can't be `true`. Why don't we use `null` as we're doing with other options?

Reply:
null is a fair option, I suggested `false`, since that's what I did with Vue: https://github.com/algolia/vue-instantsearch/blob/master/src/mixins/widget.js#L47-L55

EDIT: it's actually `true` there, but `null` would make more sense (maybe a next major)

Reply:
Yeah, `null` sounds good.
https://github.com/algolia/instantsearch.js/pull/4446/commits/b0b7add67b28f14da18f9e2d30c0b57649588567